pkg/cri/store/container/status.go

```diff
@@ -97,6 +97,62 @@ type Status struct {
 	// Unknown indicates that the container status is not fully loaded.
 	// This field doesn't need to be checkpointed.
 	Unknown bool `json:"-"`
+
+	// LinuxResources has container runtime resource constraints
+	LinuxResources *LinuxResources `json:"linuxResources,omitempty" platform:"linux"`
+	// WindowsResources has container runtime resource constraints for containers running on Windows.
+	WindowsResources *WindowsResources `json:"windowsResources,omitempty" platform:"windows"`
+}
+
+// LinuxResources has container runtime resource constraints
+type LinuxResources struct {
```

Comment:
should either use or point to the CRI definitions as the source of these, as they are a mirror of each other and I'm guessing we want them to remain a mirror?

Reply:
I just gave it a bit more thinking. I think it makes more sense to point to the CRI definition instead of keeping a mirror in [pkg/cri/store/container/status.go](https://github.com/containerd/containerd/pull/6517/files/ca1d45c6c874519e6ae6b11dffd1701df38ab6e7#diff-c2be5fd4b479b8b7191b7800ba25be98107c143e0a13575917220a4bfebe132a). At least we will have cleaner code this way.

In future if more fields are added to CRI's Linux/WindowsContainerResources, containerd needs to update the copyResourcesToStatus() function anyway, so containerd will still have control over what fields are returned to the CRI API caller.

I will update the PR to point to CRI def.